### PR TITLE
Tic - Tac - Toe: Milestone: Game logic

### PR DESCRIPTION
**Pull request by KennyUTN**
Tic - Tac - Toe: Milestone #3: Game Logic.

In this part of the project, I did everything related to handling the logic of the game.

Main advances were:
 
- Tracking the moves of each player
- Tracking the board state after each turn
- Validating legal/illegal moves.

### CODE REVIEW

**Comment by lekegitrepo** [comment]
Hi,
I'm going to mark this pull request invalid because of the linter error that is not fixed. Please, fix this error and submit it for another review.

**Comment by KennyUTN** [comment]
hey! You are correct!. I forgot to lint a little small .rb file. Thanks for
the quick heads up!

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Invalid :red_circle:   
### Hello  @KennyUTN  :wave: 

This code review request is marked invalid as your linters are failing, please make sure your linters pass and your project can be executed correctly before submitting for a new code review. 

#### If you have any question kindly reach me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (TSE)
> GitHub: [@bini-i](https://github.com/bini-i)
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

**Review by smcommits — CHANGES_REQUESTED** [review]
# Status: Changes Required
## Hello, team! You have done a great job. The application appears to be good.  But, you still need to address some significant issues. 

## Required Changes

- [ ] **Game  should inform the player if the selected move is invalid** 
Currently, you are checking if a move is invalid but you are not informing a player that they have made an invalid move. Please fix this issue. 

- [ ] **No puts/print and gets.chomp outside main.rb file** 
 main.rb should be responsible for the user interaction. This means that there should be now puts/prints or gets.chomp outside of main.rb. Please take out `puts` statement from your Logic class

- [ ] **Your Main class should contain only methods and properties.**
A class in OOP is made up of operations ( methods ) and properties ( instance variables ). Your Main class contains more than 100 lines of loose code. This defeats the purpose of object-oriented design. Please refactor every verb or logic in your Main class into their own instance methods. 
Please refer to [this quick guide](https://www.tutorialspoint.com/ruby/ruby_object_oriented.htm) on OOP in Ruby. 

- [ ] **All game-logic related resources (classes and/or modules) are kept in the lib directory**
Apart from refactoring the Main class please make sure to put it in its own file inside of the lib folder. All of the application logic should reside in the lib folder. This is the basic structure of Ruby programs.

- [ ] **Please try to write a more descriptive PR message***
Your PR message should briefly talk about your work in a particular PR. 


#### Please note that you will be making some significant changes in your source code. Therefore, the next TSEs might require some additional changes based on those. 

## Optional
#### Note: The following changes are optional 

 -  [ ] **The Classes should represent the different objects of your application.** 
 Your classes should represent different objects of your application. You should name them and such and always remember to follow the Single Responsibility Principle. For instance, rather than having a class named `logic`, you can have a class called `Board`, or rather than having a class named `Main`, you can have a class called `Game`. This way, the code will be easy to understand and maintain. You can put all the gameplay-related logic in the `Game` class and board-related logic in the `Board` class. 

# 
### Please don't hesitate to contact me on [Slack](https://microverse-students.slack.com/team/U017N7LLZ40) if you need assistance or believe I have overlooked something and made an unfair request.

File: bin/main.rb [hunk 1]
```diff
@@ -1,82 +1,152 @@
...
+        puts ''
+
+        puts 'Choose your desired available number'
+        move = gets.chomp.to_i
+
+        while (1..9).none?(move) || @board[move - 1] == 'X' || @board[move - 1] == 'O'
+
+          puts 'Illegal Move, Choose an available number between 1 and 9'
+          move = gets.chomp.to_i
+        end
```
Comment (bini-i) [inline bin/main.rb]: make sure to include checking valid input inside lib folder in its corresponding class.

File: bin/main.rb [hunk 2]
```diff
@@ -1,82 +1,152 @@
...
+
+        @game_logic.updt_board(move, 'O') if @game_logic.turn.even?
+
+        @game_logic.updt_board(move, 'X') if @game_logic.turn.odd?
+        @game_logic.turn += 1
+
+        # puts "Debug: Player 1 Moves #{@game_logic.player1_array}"
+        # puts "Debug: Player 2 Moves #{@game_logic.player2_array}"
+        # puts "Debug: Board #{@board}"
+        # puts "Turn Number #{@game_logic.turn - 1}"
```
Comment (bini-i) [inline bin/main.rb]: make sure to remove comments from commited code

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes requred :yellow_circle:  
### Hello  team :wave: 

Your project is looking good and thanks for working on the requirements suggested by the previous TSE but you have to work on all of them for the project to be approved:
- Please include all game logics in lib folder. 
  eg: checking valid inputs is inside main.rb (refer to comment below)
- Make sure to write a descriptive README (write about the features introduced in the pull request)
- Remove inline comments from committed code (see comment below)

#### If you have any question kindly reach me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (TSE)
> GitHub: [@bini-i](https://github.com/bini-i)
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @KennyUTN 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes.
- [ ] Please include all game logics in `lib` folder. (As previous TSE suggested)

##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: 
----------
Hello @KennyUTN You did an amazing job in this milestone :muscle: I really like how hard you work on the required changes and in this project :clap: 
------------------
Now you can merge this branch :+1: :1st_place_medal: 
Reachme on SLACK if you have any questions.
@ Brian Sammit Cruz Rodriguez
CONGRATULATIONS!!! AND HAPPY CODING :smile: 

well-1

**Pull request merged**